platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventIntake.java

```diff
@@ -143,6 +171,16 @@ public void addEvent(final EventImpl event) {
                         event::toMediumString);
             }
         }
+
+        if (prehandlePool == null) {
+            // Prehandle transactions on the intake thread (i.e. this thread).
+            prehandleEvent.accept(event);
+        } else {
+            // Prehandle transactions on the thread pool.
+            prehandlePool.submit(buildPrehandleTask(event));
+            event.signalPrehandleCompletion();
```

Comment:
Are you calling `event.signalPrehandleCompletion()` here intentionally? Looks like it is being called in the runnable returned from `buildPrehandleTask`.

Maybe you were you intending to call `signalPrehandleCompletion` in the block above, when `prehandlePool` is `null`?

Reply:
I've completely removed the call to `event.signalPrehandleCompletion();`. This was an artifact of me moving a bunch of code around (incorrectly), and not intentional. The only place we need to signal completion is at the end of the runnable that gets submitted to the thread pool.